README.md

```diff
@@ -112,47 +114,103 @@ trajectoryTimestamped:1:1:1627551892437;2:2:1627551892448;3:3:1627551892459|xy
+widget label, they will automatically be displayed on the same widget.
+The text before the comma will be considered as the name of the shape telemetry.
 
-For a cube, **B** should be as follow : 
-{
-    "shape" : "cube",
-    "position" : {"x": _ , "y": _ , "z" : _ },
-    "rotation" : {"x": _ , "y": _ , "z" : _ },
-    "height" : _ ,
-    "width" : _ ,
-    "depth" : _ ,
-}
+If **A** doesn't contain a comma, its whole text will be considered as the name of the shape telemetry.
+
+#### Writing **C** (the text representing the shape)
+
+**C** is built as a concatenation of the shape properties followed by their values, the whole with colons in between.
+
+LIST OF PROPERTIES : 
+
+- "shape" or "S"  => the shape type (either "cube" or "sphere" for the moment).
+
+- "position" or "P" => the position of the center of the sphere in a cartesian coordinate system 
+    1st argument : x, 2nd argument : y, 3rd argument : z
+    
+- "rotation" or "R" => the rotation of the shape using Euler angles ( please avoid this method and use a quaternion instead )
+    1st argument : the rotation around the x axis, 2nd argument : the rotation around the y axis, 3rd argument : the rotation around the z axis
+    
+- "quaternion" or "Q" => the rotation of the shape using a quaternion
+    1st argument : x coordinate, 2nd argument : y coordinate, 3rd argument : z coordinate, 4th argument : w coordinate
+    
+- "color" or "C" => the color of the shape, ex : "blue", "#2ecc71" ... 
+
+- "opacity" or "O" => the opacity of the shape, float between 0 and 1, 0 being fully transparent and 1 fully opaque ( set to 1 by default ) 
+
+=== Sphere only ===
+
+- "precision" or "PR" => the number of rectangles used to draw a sphere (the bigger the more precise, by default = 15)
+
+- "radius" or "RA"=> the radius of the sphere
+
+=== Cube only ===
+
+- "height" or "H" => the height of the cube
+- "width" or "W" => the width of the cube
+- "depth" or "D" => the depth of the cube
+
+
+
+If you don't want to send all the arguments of a certain property, you still have to add colons ( ex : position::-1:1, here the x position is not given but colons are still present )
+
+for unspecified properties, teleplot will use the ones from the previous shape state.
+
+If it is the first time teleplot receives data of a certain shape, the property "shape" (cube or sphere) has to be given and
+missing properties will be replaced by default ones.
+
+Also, the shape, color and precision properties can not be changed later on.
+
+#### Some examples
+Creating a simple sphere and cube : 
+
+    cube without timestamp :
+    - `3D|mySimpleCube:S:cube:P:1:1:1:R:0:0:0:W:2:H:2:D:2:C:#2ecc71`
+
+    sphere with timestamp : 
+    - `3D|mySimpleSphere:1627551892437:S:sphere:P::2::RA:2:C:red`
+
+Creating a cube that grows and rotates : 
+
+    In the first request we send, we need to specify the shape:
+    - `3D|my_super_cube:S:cube:W:1:D:1:H:1:C:blue`
+
+    Then we specified only properties that change :
```

Comment:
While we can, we might not recommend to only send changed properties as this will only work if teleplot was open before the the streaming script. For exemple, a refresh of the teleplot page in the navigator will result in a not displayed shape.

Reply:
up